src/ocean/text/xml/Document.d

```diff
@@ -260,6 +271,7 @@ else
 }
                 newlist;
                 index = 1;
+                node_text_buff.clear();
```

Comment:
Clearing concat buffer only when whole document is reset is not a very memory friendly pattern :(

Reply:
I may be misreading the code, but I think that's no different to what happens already. This is the only place where nodes ever get added back to the freelist, as far as I can tell.

Reply:
AFAIR in your previous version there was a fixed buffer per field which got overwritten each time `name` was called (invalidating old references).